src/app/api/cronjob/route.ts

```diff
@@ -0,0 +1,94 @@
+import type { NextRequest } from "next/server";
+import { NextResponse } from "next/server";
+import {
+  addDays,
+  eachDayOfInterval,
+  format,
+  isWithinInterval,
+  subDays,
+} from "date-fns";
+import { and, eq, gte, lte } from "drizzle-orm";
+
+import { db } from "@/lib/db";
+import {
+  gameSessions,
+  gameSessionSchedules,
+  semesters,
+  weekdayEnum,
+} from "@/lib/db/schema";
+import { insertGameSessionSchema } from "@/lib/validators";
+
+export async function POST(req: NextRequest) {
+  try {
+    const now = new Date();
+    const oneWeekFromNow = addDays(now, 7);
+
+    const activeSemester = await db.query.semesters.findFirst({
+      where: and(
+        lte(semesters.startDate, format(now, "yyyy-MM-dd")),
+        gte(semesters.endDate, format(now, "yyyy-MM-dd")),
+      ),
+    });
+
+    if (!activeSemester) {
+      return NextResponse.json("No active semester found.", { status: 400 });
+    }
+
+    if (
+      isWithinInterval(now, {
+        start: activeSemester.breakStart,
+        end: activeSemester.breakEnd,
+      })
+    ) {
+      return NextResponse.json("Currently in semester break period.", {
+        status: 400,
+      });
+    }
+
+    const daysOfWeek = eachDayOfInterval({ start: now, end: oneWeekFromNow });
```

Comment:
We want to guarantee that each game session generated is in the semester. What happens if the script is ran on the last day of the semester?

Reply:
yeah, that would be a problem, but I assumed it would be run on every Monday.

Reply:
Also should I make it so that the script generates sessions for the next 7 days (so not including the current day)? So if the script was run on a sunday then it will generate all sessions for the next monday - sunday?

Reply:
Honestly the implementation of the script is a bit fuzzy for me too so I would love to hear some of your own insights.

I did some thinking about this and think the best window for the script to generate events is the 7 day window starting from the next bookingOpenDay.

For example, if the script is called every Wednesday, and the bookingOpenDay is on a Tuesday, then we would generate all the events starting from next Tuesday to the Monday after (inclusive).
i.e If the script is called `2024-06-12`, it will generate events inclusively from `2024-06-18` to `2024-06-24`, while making sure that all of those events are inside the semester date range and outside the break range.